### PR TITLE
Remove unused plugins from weekly.ci

### PR DESCRIPTION
**Pull request by NotMyFault**
Thanks to Hervé's great work, we can remove a few more plugins we don't use on weekly.ci.

### CODE REVIEW

File: plugins-weekly.ci.jenkins.io.txt [hunk 1]
```diff
@@ -26,7 +25,6 @@ coverage:1.7.0
 command-launcher:107.v773860566e2e
 commons-lang3-api:3.13.0-62.v7d18e55f51e2
 commons-text-api:1.11.0-95.v22a_d30ee5d36
-config-file-provider:959.vcff671a_4518b_
```
Comment (dduportal) [inline plugins-weekly.ci.jenkins.io.txt:29]: Note that this plugin is still installed as it is a dependency of `job-dsl`:

```
[2024-01-09T18:47:08.571Z] #23 40.62 job-dsl depends on: 
[2024-01-09T18:47:08.571Z] #23 40.62 configuration-as-code 1646.v98b_72d209775
[2024-01-09T18:47:08.571Z] #23 40.62 branch-api 2.1105.v472604208c55
[2024-01-09T18:47:08.571Z] #23 40.62 cloudbees-folder 6.815.v0dd5a_cb_40e0e
[2024-01-09T18:47:08.571Z] #23 40.62 config-file-provider 938.ve2b_8a_591c596
```

File: plugins-weekly.ci.jenkins.io.txt [hunk 2]
```diff
@@ -41,7 +39,6 @@ docker-commons:439.va_3cb_0a_6a_fb_29
 durable-task:523.va_a_22cf15d5e0
 echarts-api:5.4.3-2
 extended-read-permission:53.v6499940139e5
-favorite:2.208.v91d65b_7792a_c
```
Comment (dduportal) [inline plugins-weekly.ci.jenkins.io.txt:44]: Note that this plugin is a dependency of `customizable-header`:

```[2024-01-09T18:47:08.050Z] #23 40.15 customizable-header depends on: 
[2024-01-09T18:47:08.051Z] #23 40.15 caffeine-api 3.1.8-133.v17b_1ff2e0599
[2024-01-09T18:47:08.051Z] #23 40.15 ionicons-api 56.v1b_1c8c49374e
[2024-01-09T18:47:08.051Z] #23 40.15 antisamy-markup-formatter 162.v0e6ec0fcfcf6
[2024-01-09T18:47:08.051Z] #23 40.15 cloudbees-folder 6.858.v898218f3609d
[2024-01-09T18:47:08.051Z] #23 40.15 favorite 2.208.v91d65b_7792a_c
```

**Review by dduportal — APPROVED** [review]
Approving but please note that 2 of the removed plugins are still installed as transitive dependencies (see comment for details).

Not a blocker, but it means these plugin will have their version downgraded until the parent dependency bumps to a recent version.

I would keep these plugins in here to ensure it's always the latest version, WDYT @NotMyFault ?

File: plugins-weekly.ci.jenkins.io.txt (hunk 1 above)
Comment (NotMyFault) [reply plugins-weekly.ci.jenkins.io.txt:29]: config-file-provider is an optional dependency only, https://github.com/jenkinsci/job-dsl-plugin/blob/fb7d49a2d5a002bf82449cdf96c4b98ea8a630ae/job-dsl-plugin/pom.xml#L103

Optional dependencies are installed when needed, but given that the plugin is unistallable from the web ui, job-dsl doesn't hard depend on it:
Screenshot 2024-01-10 at 09 46 11

File: plugins-weekly.ci.jenkins.io.txt (hunk 2 above)
Comment (NotMyFault) [reply plugins-weekly.ci.jenkins.io.txt:44]: Same like above

**Comment by NotMyFault** [comment]
> I would keep these plugins in here to ensure it's always the latest version, WDYT @NotMyFault ?

No objections, see the comment at the first plugin for the reasoning of a potential removal 😅

**Comment by dduportal** [comment]
> config-file-provider is an optional dependency only, https://github.com/jenkinsci/job-dsl-plugin/blob/fb7d49a2d5a002bf82449cdf96c4b98ea8a630ae/job-dsl-plugin/pom.xml#L103
> 
> Optional dependencies are installed when needed, but given that the plugin is unistallable from the web ui, job-dsl doesn't hard depend on it: Screenshot 2024-01-10 at 09 46 11

Interesting, thanks for the explanation @NotMyFault . It means the [`jenkins-plugin-cli`](https://github.com/jenkinsci/plugin-installation-manager-tool) install optional dependencies. I wasn't able to (quickly) find a flag in this CLI to decide wether or not to install optional dependencies.

Do you agree on this (I'm asking because I don't know well all details)?

**Comment by NotMyFault** [comment]
> > config-file-provider is an optional dependency only, https://github.com/jenkinsci/job-dsl-plugin/blob/fb7d49a2d5a002bf82449cdf96c4b98ea8a630ae/job-dsl-plugin/pom.xml#L103
> 
> > 
> 
> > Optional dependencies are installed when needed, but given that the plugin is unistallable from the web ui, job-dsl doesn't hard depend on it: Screenshot 2024-01-10 at 09 46 11
> 
> 
> 
> Interesting, thanks for the explanation @NotMyFault . It means the [`jenkins-plugin-cli`](https://github.com/jenkinsci/plugin-installation-manager-tool) install optional dependencies. I wasn't able to (quickly) find a flag in this CLI to decide wether or not to install optional dependencies.
> 
> 
> 
> Do you agree on this (I'm asking because I don't know well all details)?

For sure! Interestingly tho. I'll take a look later and file a RFE to separate the list between optional and hard dependencies

**Pull request merged**